src/burnchains/db.rs

```diff
@@ -127,10 +207,45 @@ CREATE TABLE burnchain_db_block_ops (
     block_hash TEXT NOT NULL,
     op TEXT NOT NULL,
     txid TEXT NOT NULL,
+
     FOREIGN KEY(block_hash) REFERENCES burnchain_db_block_headers(block_hash)
 );
 
-CREATE TABLE db_config(version TEXT NOT NULL);";
+CREATE TABLE affirmation_maps (
+    affirmation_id INTEGER PRIMARY KEY AUTOINCREMENT,
+    weight INTEGER NOT NULL,
+    affirmation_map TEXT NOT NULL
+);
+
+-- ensure anchor block uniqueness
+CREATE TABLE anchor_blocks (
+    reward_cycle INTEGER PRIMARY KEY  -- will be i64::MAX if absent
+);
+
+CREATE TABLE block_commit_metadata (
+    burn_block_hash TEXT NOT NULL,
+    txid TEXT NOT NULL,
+    block_height INTEGER NOT NULL,
+    vtxindex INTEGER NOT NULL,
+    
+    affirmation_id INTEGER NOT NULL,
+    anchor_block INTEGER NOT NULL,
```

Comment:
Does it make more sense for this to be nullable rather than using i64::max to indicate null? That way we wouldn't have to manually convert the option type for the field.

Reply:
`anchor_block` is a `FOREIGN KEY` to the `anchor_blocks` table, so it can't be `NULL`.  The `FOREIGN KEY` constraint is in place to ensure that only one anchor block is chosen for a given reward cycle, even though many block commits can have the same anchor block.

Reply:
Is it true that a `FOREIGN KEY` constrained column cannot be null?

```
$ sqlite3 /tmp/test.db
SQLite version 3.32.3 2020-06-18 14:16:19
Enter ".help" for usage hints.
sqlite> PRAGMA foreign_keys = ON;
sqlite> CREATE TABLE parent(a INTEGER PRIMARY KEY);
sqlite> CREATE TABLE child(a INTEGER PRIMARY KEY, b INTEGER, FOREIGN KEY(b) REFERENCES parent(a));
sqlite> INSERT INTO child VALUES (0, 1);
Error: FOREIGN KEY constraint failed
sqlite> INSERT INTO child VALUES (0, NULL);
sqlite> SELECT * FROM child;
0|
```

Reply:
Huh, well look at that. Knowledge updated.

Reply:
Fixed in fb4d7f9